### DISCUSSION TITLE
Replacing SignalR with RabbitMQ and MQTT or STOMP

### DISCUSSION THREAD

**Discussion by mlsomers** [rabbitmq/rabbitmq-dotnet-client#1282] [Q&A] [answered]
I am considering to replace our SignalR setup with a RabbitMQ solution, using the MQTT or STOMP plugin. The reason being that SignalR is not cross-compatible over framework versions and platforms (.Net framework and .Net Core variants cannot communicate with each other).

Before diving in I want to be sure that one important requirement is met.

We need to keep track of the number of concurrent clients or logins per account, and automatically logoff the least-recent connected client when the quota exceeds a maximum. The maximum will be different for each user.

RabbitMQ does not need to do this out of the box, but it must at least provide the info so we can implement it by sending the client appropriate messages (to automatically log off or unsubscribe)...

Can RabbitMQ provide connect/disconnect event info to our server by filing a message into a server-queue whenever a client connects or disconnects from a different set of queues?

**Comment by lukebakken** [comment, marked as answer]
Seems like the event exchange should provide what you need - https://www.rabbitmq.com/event-exchange.html

Since you're using MQTT or STOMP I'm not 100% sure if the events will provide the username. You'll have to give that a try. If they don't, let us know and maybe it can be added.

**Reply by mlsomers** [reply]
Thanks! this is looking good so far!
